crates/common/src/fuzz/mod.rs

```diff
@@ -211,55 +229,59 @@ impl FuzzRunner {
                 .collect::<Vec<String>>(),
         );
 
-        let fork = Arc::new(self.forker.clone()); // Wrap in Arc for shared ownership
         let dotrain = Arc::new(self.dotrain.clone());
         let mut handles = vec![];
 
-        for _ in 0..no_of_runs {
-            let fork_clone = Arc::clone(&fork); // Clone the Arc for each thread
-            let elided_binding_keys = Arc::clone(&elided_binding_keys);
-            let deployer = Arc::clone(&deployer);
-            let scenario_bindings = scenario_bindings.clone();
-            let dotrain = Arc::clone(&dotrain);
-
-            let mut final_bindings: Vec<Rebind> = vec![];
-
-            // for each scenario.fuzz_binds, add a random value
-            for elided_binding in elided_binding_keys.as_slice() {
-                let mut val: [u8; 32] = [0; 32];
-                self.rng.fill_bytes(&mut val);
-                let hex = format!("0x{}", alloy_primitives::hex::encode(val));
-                final_bindings.push(Rebind(elided_binding.to_string(), hex));
-            }
+        for block_number in blocks {
+            self.forker.roll_fork(Some(block_number), None)?;
+            let fork = Arc::new(self.forker.clone()); // Wrap in Arc for shared ownership
+
+            for _ in 0..no_of_runs {
+                let fork_clone = Arc::clone(&fork); // Clone the Arc for each thread
+                let elided_binding_keys = Arc::clone(&elided_binding_keys);
+                let deployer = Arc::clone(&deployer);
+                let scenario_bindings = scenario_bindings.clone();
+                let dotrain = Arc::clone(&dotrain);
+
+                let mut final_bindings: Vec<Rebind> = vec![];
+
+                // For each scenario.fuzz_binds, add a random value
+                for elided_binding in elided_binding_keys.as_slice() {
+                    let mut val: [u8; 32] = [0; 32];
+                    self.rng.fill_bytes(&mut val);
+                    let hex = format!("0x{}", alloy_primitives::hex::encode(val));
```

Comment:
does this ensure even number of hex nibbles on the output? i.e. left pad with 0 til even number of chars

Reply:
i've added padding to the format string